app/javascript/packs/leafletMap.js

```diff
@@ -149,3 +160,52 @@ function recalculateTooltipVisibility() {
         }
     });
 }
+
+const syncUserPositionWithServerImpl = async (location) => {
```

Comment:
This and everything else regarding the geolocation should get it's own js file because it's not part of the map. You can then import the `addMarker `method there to show your marker on the map